main.c

```diff
@@ -1057,14 +1061,27 @@ int main(int argc, char **argv)
 	if (tcmu_setup_log())
 		goto destroy_config;
 
-	tcmu_dbg("handler path: %s\n", handler_path);
+	fd = creat(TCMU_LOCK_FILE, S_IRUSR | S_IWUSR);
+	if (fd == -1) {
+		tcmu_err("creat(%s) failed: [%m]\n", TCMU_LOCK_FILE);
+		goto destroy_log;
+	}
+
+	lock.l_type = F_WRLCK;
+	if (fcntl(fd, F_SETLK, &lock) == -1) {
+		tcmu_err("tcmu-runner is already running...\n");
+		close(fd);
```

Comment:
We'd better add one goto for closing the fd to make the code more readable.

Or we should it close(fd) everywhere just after this line when trying to goto error tags.

Reply:
Good catch @lxbsz